cmake/Modules/FunctionQbtCXXCompilerAndModeCheck.cmake

```diff
@@ -0,0 +1,19 @@
+function(qbt_cxx_compiler_and_mode_check _min_std)
+    # ensure the compiler fully supports the minimum required C++ mode.
+    if(NOT CMAKE_CXX${_min_std}_STANDARD__HAS_FULL_SUPPORT)
+        message(FATAL_ERROR "${PROJECT_NAME} requires a compiler with full C++${_min_std} support")
+    endif()
+
+    # now we know that the compiler fully supports the minimum required C++ mode,
+    # but we must still prevent the user or compiler from forcing/defaulting to an insufficient C++ mode
+    if(NOT CMAKE_CXX_STANDARD)
+        set(CMAKE_CXX_STANDARD ${_min_std} PARENT_SCOPE)
+    elseif((CMAKE_CXX_STANDARD VERSION_LESS ${_min_std}) OR (CMAKE_CXX_STANDARD VERSION_EQUAL 98))
+        message(FATAL_ERROR "${PROJECT_NAME} has to be built with at least C++${_min_std} mode.")
+    endif()
+
+    set(CMAKE_CXX_STANDARD_REQUIRED ON PARENT_SCOPE)
+
+    message(STATUS "Building in C++${CMAKE_CXX_STANDARD} mode.\n"
```

Comment:
In travis it prints out "-- Building in C++ mode."

Reply:
@zeule fixed. I moved the informational message to top level because it seemed like the easiest and most obvious way to get it to print correctly regardless of whether or not the user passes the `-DCMAKE_CXX_STANDARD` option in the command line.